### DISCUSSION TITLE
New voice for each chapter based on chapter title

### DISCUSSION THREAD

**Discussion by the-lilith-cluster** [aedocw/epub2tts#226] [General]
Is it possible for each chapter to have a unique voice? I have a book that is first person POV with a rotating cast, and the characters' names are in the chapter title. Having each of those be a unique voice would work really well for it.

**Comment by aedocw** [comment]
This is a pretty interesting idea, and I think could be done pretty easily at chapter breaks the way I'm thinking about implementing it.

I'll also have to think carefully about which engines can support this, though even that might not be a big challenge. It would definitely be easy with edge, maybe be a little more tricky with coqui voices because of where/how the model and speaker object is declared. Though it probably will not be much of a performance penalty to move that to a place that could support changing the speaker as needed.

I think including `%speaker` at the start of a chapter would allow one to specify which speaker, and would be easy to put in as part of the initial book scan. Something like:

```
# Chapter one
%en-US-AvaNeural

Blah blah blah

# Chapter two
%en-US-AndrewNeural

More things to say
```

Not sure when I'll have time to work on this but it will probably be within a week or two. Anyone else who wants to take a shot at this and submit a PR is welcome to :)

**Comment by aedocw** [comment]
If you get a chance to try out the latest release that supports this, please let me know if it worked out for you. Thanks!

**Reply by 4rc** [reply]
How do you declare coqui voices with these method? i tried some
 # Part 4 % it-IT-"Brenda Stern"
 # Part 4 % it-IT-BrendaStern
 # Part 4 % it-IT-Brenda Stern
 # Part 4 % it-IT Brenda Stern

but none worked
this is probably a very dumb question, but I did not find any reference...
thanks

**Reply by aedocw** [reply]
I'm not certain it will work properly with coqui voices (I can't remember what I tested), but the line has to start with "#". Try a line that looks like this:
`# Part 4 % it-IT Brenda Stern`

**Reply by 4rc** [reply]
yes, sorry, i had used the hashtag in the lines I posted, they transformed the text in big&bold. I actually already tried the line you suggested. Probably this is not possible at the time. thanks!
